nilearn/plotting/displays.py

```diff
@@ -237,7 +245,28 @@ def transform_to_2d(self, data, affine):
                 The affine of the volume
 
         """
-        max_axis = 'xyz'.index(self.direction)
+        if self.direction in 'xlr':
+            max_axis = 0
+        else:
+            max_axis = '.yz'.index(self.direction)
+
+        # set unselected brain hemisphere activations to 0
+
+        if self.direction == 'l':
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection = data[int(x_center):, :, :]
+        elif self.direction == 'r':
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection = data[:int(x_center), :, :]
+        else:
+            data_selection = data
```

Comment:
The code above will only work if the affine is diagonal. Are we sure that it is diagonal at this point?